doc/design-sparse.md

```diff
@@ -0,0 +1,52 @@
+# Sparse dimension support
+
+In our case, the main application of sparse dimensions is neutron event data.
+So far we have assumed that a dataset is not aware of the dimension of the event data, since it has a different length at every point.
+For example, event data could be stored as a nested dataset for each of the points in the volume of the outer dataset (see https://github.com/scipp/scipp/blob/master/scippy/demo/demo-part3.ipynb for an example).
+This works, but it has the drawback of making handling such data different and more cumbersome, since we must access nested structures and provide special routines.
+
+Here we discuss another option, supporting sparse data by making `Dataset` and `Variable` compatible with sparse dimensions.
+This could work as follows:
+
+1. There can be only a single sparse dimension per `Variable` or `Dataset`, and it must be the innermost dimension.
+
+2. `class Dimensions` would not store the extent of the sparse dimension, but an invalid extent (e.g., -1). For display, we simply print "sparse", not "-1".
+
+3. Storing event data:
+   - `Coord.Tof`, `Coord.Energy`, ... used for, e.g., time-of-flight, .... This removes the need for having a number of special tags for events, such as `Data.EventTof`, `Data.EventEnergy`, ..., which are required in the current model without sparse-dimension support.
+   - `Coord.PulseTime` for pulse times.
+   - `Data.Value` and `Data.Variance` for weights. These can be omitted for unweighted events.
```

Comment:
I'm not clear on how `Data.Variance` will relate to weights?

Reply:
It is the equivalent to `Mantid::DataObjects::WeightedEvent::m_errorSquared`